changelog.d/16052.bugfix

```diff
@@ -0,0 +1 @@
+Linearize push rule modification endpoints to prevent deadlocks. Contributed by Nick @ Beeper (@fizzadar).
```

Comment:
```suggestion
Fix long-standing bug where concurrent requests to change a user's push rules could cause a deadlock. Contributed by Nick @ Beeper (@fizzadar).
```

Reply:
Err, can you commit that please as I don't have permissions too :(

Reply:
Done! Silly org permission thing means we can't enable that either apparently!